libcaja-private/caja-open-with-dialog.c

```diff
@@ -247,7 +247,7 @@ add_or_find_application (CajaOpenWithDialog *dialog)
         }
     }
 
-    if (app == NULL)
+    if ((app == NULL) && (error))
```

Comment:
that's cute, but normally `app` can only be `NULL` when `error` is set, so it's kind of pointless.

Also, this prevents the `return` inside this `if` it `error` was *really* not set, which leads to more bugs below, like using a `NULL` `app` as parameter to [`g_app_info_set_as_default_for_type()`](https://developer.gnome.org/gio/stable/GAppInfo.html#g-app-info-set-as-default-for-type) and friends.

If you really want to guard accessing `error`, I'd rather do that in the if's body, like `error ? error->message: _("unknown error")`.